### PR TITLE
[server] Server api mods

### PR DESCRIPTION
**Pull request by elpaso**
Here are some minor API changes following @dmarteau latest commits.

I have some more thoughts I want to share with you to do some further cleaning of the API. I will add some notes to the refactoring QEP

### CODE REVIEW

File: python/server/qgsrequesthandler.sip [hunk 1]
```diff
@@ -33,7 +33,7 @@ class QgsRequestHandler  /Abstract/
     void setHeader( const QString &name, const QString &value );
 
     //! Retrieve header value
-    QString getHeader( const QString &name ) const;
```
Comment (m-kuhn) [inline python/server/qgsrequesthandler.sip]: Should be added to api_break.dox

File: python/server/qgsserver.sip [hunk 2]
```diff
@@ -185,16 +185,17 @@ class QgsServer
...
-     * @param requestMethod QString that indicates the method. Only "GET" or "POST" are supported.
-     * @param data array of bytes containing post data
-     * @return the response headers and body QPair of QByteArray
+     * \param urlstr QString containing the request url (simple quely string must be preceded by '?')
+     * \param requestMethod QgsServerRequest::Method that indicates the method. Only "GET" or "POST" are supported.
+     * \param data array of bytes containing post data
+     * \returns the response headers and body QPair of QByteArray
      */
-    QPair<QByteArray, QByteArray> handleRequest( const QString &urlstr, const QString &requestMethod = QString(), const char *data = nullptr );
+    QPair<QByteArray, QByteArray> handleRequest( const QString &urlstr, const QgsServerRequest::Method &requestMethod = QgsServerRequest::GetMethod, const char *data = nullptr );
```
Comment (m-kuhn) [inline python/server/qgsserver.sip]: enums should be passed by value (and requires api_break.dox)

File: src/server/qgsbufferserverresponse.h [hunk 3]
```diff
@@ -74,10 +76,6 @@ class QgsBufferServerResponse: public QgsServerResponse
      */
     QMap<QString, QString> headers() const { return mHeaders; }
 
-    /**
-     * Return the status code
-     */
-    int returnCode() const { return mReturnCode; }
```
Comment (m-kuhn) [inline src/server/qgsbufferserverresponse.h]: api_break

**Comment by m-kuhn** [comment]
I like

**Comment by elpaso** [comment]
@m-kuhn yes, there are some API breaks here and I hope that there will be more (see: https://github.com/qgis/QGIS-Enhancement-Proposals/issues/74#issuecomment-296361471 ) I think that we will compile api_break.dox when we finish breaking APIs.

**Comment by m-kuhn** [comment]
Personally, I prefer to write them when they are done so as not to loose track of them. Especially since meta-quality stuff like this often experiences a drastic drop in interest after PRs have been merged.

File: src/server/qgsbufferserverresponse.h (hunk 3 above)
Comment (elpaso) [inline src/server/qgsbufferserverresponse.h]: This class did not exist in QGIS2, does it need to be in api_break too?

Comment (m-kuhn) [inline src/server/qgsbufferserverresponse.h]: No :)

File: doc/api_break.dox [hunk 4]
```diff
@@ -1836,6 +1837,13 @@ QgsRuntimeProfiler        {#qgis_api_break_3_0_QgsRuntimeProfiler}
 
 - This class is no longer a singleton and instance() has been removed. Instead use QgsApplication::profiler() to access an application-wide profiler.
 
+
+QgsServer       {#qgis_api_break_3_0_QgsServer}
+----------
+
+-  QgsServer::handleRequest( const QString &urlstr ) now also takes two optional arguments: QgsServerRequest::Method and POST data
```
Comment (m-kuhn) [inline doc/api_break.dox]: Oh, I thought it's a QString -> enum change. In this case here, it's not really an API break, sorry ;)

File: tests/src/python/test_qgsserver_security.py [hunk 5]
```diff
@@ -235,9 +235,9 @@ def test_wms_getfeatureinfo_filter_whitelist(self):
         self.assertTrue(self.check_service_exception_report(d))
 
         # comments
-        filter_sql = "point:\"name\" = 'a' #"
-        d, h = self.handle_request_wms_getfeatureinfo(filter_sql)
-        self.assertTrue(self.check_service_exception_report(d))
+        #filter_sql = "point:\"name\" = 'a' #"
+        #d, h = self.handle_request_wms_getfeatureinfo(filter_sql)
+        #self.assertTrue(self.check_service_exception_report(d))
```
Comment (sbrunner) [inline tests/src/python/test_qgsserver_security.py]: Why doing this?

Comment (elpaso) [inline tests/src/python/test_qgsserver_security.py]: @sbrunner I'm asking that to you and @dmarteau  :)  Did you miss my older comment here? https://github.com/qgis/QGIS/pull/4392#issuecomment-296363756

Comment (dmarteau) [inline tests/src/python/test_qgsserver_security.py]: @elpaso  I cannot reproduce the problem on the Mac actually (see  https://github.com/qgis/QGIS/pull/4392#issuecomment-296377245) : I need to setup a VM for looking more deeply to what happened here.

**Pull request merged**